src/Set.ts

```diff
@@ -110,8 +110,15 @@ if (!has('es6-set')) {
 
 		constructor(iterable?: ArrayLike<T> | Iterable<T>) {
 			if (iterable) {
-				for (const value of iterable) {
-					this.add(value);
+				if (isArrayLike(iterable)) {
+					for (let i = 0; i < iterable.length; i++) {
+						this.add(iterable[ i ]);
```

Comment:
There are more!  It is in a few places...

Reply:
Fixed